packages/data-point/src/ReducerPath.js

```diff
@@ -39,21 +97,54 @@ function getPath(value, path) {
   return current;
 }
 
-class ReducerPath extends ReducerNative {
+/**
+ * @param {Accumulator} accumulator
+ * @param {Object} compiledPath
+ * @returns {any} resolved value
+ */
+function resolvePath(accumulator, compiledPath) {
+  const valueToCrawl =
+    compiledPath.moveUp === true ? accumulator : accumulator.value;
+
+  return crawlPath(valueToCrawl, compiledPath.path);
+}
+
+/**
+ * Reducer that crawls a data path (using dot notation) from the accumulator
+ */
+class ReducerPath extends Reducer {
+  /**
+   * @param {string} spec dot notation object path, expects path to start with $
+   */
   constructor(spec) {
     super("path", spec, spec);
-    this.path = stringToPath(spec.substr(1));
+    this.compiledPath = parsePath(spec);
   }
 
+  /**
+   * @param {string} spec
+   */
   static isType(spec) {
-    return typeof spec === "string" && spec.charAt(0) === "$";
+    return (
+      typeof spec === "string" && spec.charAt(0) === "$" && spec.length > 1
```

Comment:
the constructor should give some warning/error if this function does not return true for `spec`

Reply:
I don't believe its necessary for these reducers since they only get called by datapoint directly, for other reducers (helpers, entities) we do have to add validation with errors on constructors since they are instantiated on their own (outside of data-point)